tests/testthat/test-fit_model.R

```diff
@@ -0,0 +1,189 @@
+test_that("Minimal model fit all params runs", {
+  # Parameters
+  parameters <- list(
+    generation_interval = gostic_gt_pmf,
+    delay_interval = c(0.2, 0.8),
+    right_truncation = c(0.7, 0.3)
+  )
+  # Data -- 5 points only
+  data_path <- test_path("data", "test_data.parquet")
+  con <- DBI::dbConnect(duckdb::duckdb())
+  data <- DBI::dbGetQuery(con, "
+                         SELECT
+                           report_date,
+                           reference_date,
+                           disease,
+                           geo_value AS state_abb,
+                           value AS confirm
+                         FROM read_parquet(?)
+                         ORDER BY reference_date
+                         LIMIT 5
+                          ",
+    params = list(data_path)
+  )
+  DBI::dbDisconnect(con)
+  # Priors
+  priors <- list(
+    rt = list(
+      mean = 1,
+      sd = 0.2
+    ),
+    gp = list(
+      alpha_sd = 0.05
+    )
+  )
+  # Sampler
+  sampler_opts <- list(
+    cores = 1,
+    chains = 1,
+    adapt_delta = 0.8,
+    max_treedepth = 10,
+    iter_warmup = 100,
+    iter_samples = 100
+  )
+
+  fit <- fit_model(
+    data = data,
+    parameters = parameters,
+    seed = 12345,
+    horizon = 0,
+    priors = priors,
+    sampler = sampler_opts
+  )
+
+  expect_s3_class(fit, "epinow")
+})
+
+test_that("Minimal model fit with no right trunc or delay runs", {
+  # Parameters
+  parameters <- list(
+    generation_interval = gostic_gt_pmf,
+    delay_interval = NA,
+    right_truncation = NA
+  )
+  # Data -- 5 points only
+  data_path <- test_path("data", "test_data.parquet")
+  con <- DBI::dbConnect(duckdb::duckdb())
+  data <- DBI::dbGetQuery(con, "
+                         SELECT
+                           report_date,
+                           reference_date,
+                           disease,
+                           geo_value AS state_abb,
+                           value AS confirm
+                         FROM read_parquet(?)
+                         ORDER BY reference_date
+                         LIMIT 5
+                          ",
+    params = list(data_path)
+  )
+  DBI::dbDisconnect(con)
+  # Priors
+  priors <- list(
+    rt = list(
+      mean = 1,
+      sd = 0.2
+    ),
+    gp = list(
+      alpha_sd = 0.05
+    )
+  )
+  # Sampler
+  sampler_opts <- list(
+    cores = 1,
+    chains = 1,
+    adapt_delta = 0.8,
+    max_treedepth = 10,
+    iter_warmup = 100,
+    iter_samples = 100
+  )
+
+  fit <- fit_model(
+    data = data,
+    parameters = parameters,
+    seed = 12345,
+    horizon = 0,
+    priors = priors,
+    sampler = sampler_opts
+  )
+
+  expect_s3_class(fit, "epinow")
+})
+
+test_that("Bad params w/ failing fit issues warning and returns NA", {
```

Comment:
Can you document more clearly what part of the parameter specification is bad here?

Reply:
https://github.com/CDCgov/cfa-epinow2-pipeline/pull/26/commits/439be4e214924ab62c646676dcaefaa90857dd04